### PR TITLE
[9.4] Prevent creation of a task having an empty content; fixes #7737

### LINKED ISSUE #7737: You can create an empty task
You can create a task without any description.
You can also delete content of an existing task

### PR DESCRIPTION
**Pull request by yllen**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #7737

### CODE REVIEW

File: inc/commonitiltask.class.php [hunk 1]
```diff
@@ -229,6 +229,11 @@ function post_deleteFromDB() {
 
    function prepareInputForUpdate($input) {
 
+      if (empty($input['content'])) {
+         Session::addMessageAfterRedirect(__("You can't add a task without description"),
+                                          false, ERROR);
+         return false;
+      }
```
Comment (cedric-anne) [inline inc/commonitiltask.class.php:236]: Maybe this check should be done only if `content` is defined in input. Indeed, it should be possible to update only specific fields without having to pass any `content` field in `$input`.

File: inc/commonitiltask.class.php [hunk 2]
```diff
@@ -229,6 +229,11 @@ function post_deleteFromDB() {
 
    function prepareInputForUpdate($input) {
 
+      if (empty($input['content'])) {
+         Session::addMessageAfterRedirect(__("You can't add a task without description"),
```
Comment (cedric-anne) [inline inc/commonitiltask.class.php:233]: Message should be rewritten to be more appropriate to update context.

**Comment by cedric-anne** [comment]
I updated your branch to add a more descriptive commit message and to fix tests.

File: inc/commonitiltask.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/commonitiltask.class.php]: I fixed this.

File: inc/commonitiltask.class.php (hunk 2 above)
Comment (cedric-anne) [reply inc/commonitiltask.class.php]: I updated the message.

File: inc/commonitiltask.class.php (hunk 1 above)
Comment (yllen) [reply inc/commonitiltask.class.php]: In update, content is always in the input even you doesn't change it, so your test in not useful.
But your message in better

**Pull request merged**